borgmatic/hooks/zabbix.py

```diff
@@ -0,0 +1,93 @@
+    '''
+    Ping the configured Zabbix topic. Use the given configuration filename in any log entries.
+    If this is a dry run, then don't actually ping anything.
+    '''
+
+    run_states = hook_config.get('states', ['fail'])
+
+    if state.name.lower() in run_states:
+        dry_run_label = ' (dry run; not actually pinging)' if dry_run else ''
+
+        state_config = hook_config.get(state.name.lower(),{'value': f'invalid',},)
+
+        base_url = hook_config.get('server', 'https://cloud.zabbix.com/zabbix/api_jsonrpc.php')
+        username = hook_config.get('username')
+        password = hook_config.get('password')
+        api_key = hook_config.get('api_key')
+        itemid = hook_config.get('itemid')
+        host = hook_config.get('host')
+        key = hook_config.get('key')
+
+        value = state_config.get('value')
+
+        headers = {'Content-Type': 'application/json-rpc'}
+
+        logger.info(f'{config_filename}: Pinging zabbix {dry_run_label}')
+        logger.debug(f'{config_filename}: Using zabbix ping URL {base_url}')
+
+        # Determine the zabbix method used to store the value: itemid or host/key
+        if (itemid) is not None:
+            logger.info(f'{config_filename}: Updating {itemid} on zabbix')
+            data = {"jsonrpc":"2.0","method":"history.push","params":{"itemid":itemid,"value":value},"id":1}
+        
+        elif (host and key) is not None:
+            logger.info(f'{config_filename}: Updating Host:{host} and Key:{key} on zabbix')
+            data = {"jsonrpc":"2.0","method":"history.push","params":{"host":host,"key":key,"value":value},"id":1}
+
+        elif (host) is not None:
+            logger.warning( f'{config_filename}: Key missing for zabbix authentication' )
+
+        elif (key) is not None:
+            logger.warning( f'{config_filename}: Host missing for zabbix authentication' )
+
+        # Determine the authentication method: API key or username/password
+        auth = None
+        if (api_key) is not None:
+            logger.info(f'{config_filename}: Using API key auth for zabbix')
+            headers['Authorization'] = 'Bearer ' + api_key
+            
+        elif (username and password) is not None:
+            logger.info(f'{config_filename}: Using user/pass auth with user {username} for zabbix')
+            response = requests.post(base_url, headers=headers, data='{"jsonrpc":"2.0","method":"user.login","params":{"username":"'+username+'","password":"'+password+'"},"id":1}')
+            data['auth'] = response.json().get('result')
+
+        elif username is not None:
+            logger.warning( f'{config_filename}: Password missing for zabbix authentication, defaulting to no auth' )
+
+        elif password is not None:
+            logger.warning( f'{config_filename}: Username missing for zabbix authentication, defaulting to no auth' )
+
+        if not dry_run:
+            logging.getLogger('urllib3').setLevel(logging.ERROR)
```

Comment:
This probably needs to happen prior to your `requests` call above if it's going to have effect for that call too.

Reply:
agreed: https://github.com/borgmatic-collective/borgmatic/pull/85/commits/8edb40a8e97e40765b3b0ea1be6a1c79d5cc3edd